url/failure.html

```diff
@@ -15,6 +15,15 @@
       continue
     }
 
+    let validRelativeURL = false;
+    try {
+        new URL(test.input, window.location.href);
+        validRelativeURL = true;
+    } catch(e) { }
+    if (validRelativeURL) {
```

Comment:
I think this condition here is too restrictive, because following two tests operate on URL class and are not affected by this issue.

I'd suggest to remove this statement and add his condition to `if` statement bellow:

```diff
- if (!test.inputCanBeRelative) {
+ if (!test.inputCanBeRelative && !validRelativeURL) {
```

Reply:
Shouldn’t we instead update urltestdata.json to account for these cases, instead of trying to dynamically guess?